Build.ps1

```diff
@@ -32,149 +32,145 @@ param(
     [Parameter()]
     [switch]$IncludeJss,
     [Parameter(HelpMessage="If the docker image is already built it should be skipped.")]
-    [switch]$SkipExistingImage
+    [switch]$SkipExistingImage,
+    [Parameter()]
+    [switch]$IncludeExperimental
 )
 
+if ([string]::IsNullOrEmpty($InstallSourcePath))
+{
+    $InstallSourcePath = (Join-Path -Path $PSScriptRoot -ChildPath "\packages")
+}
+
 $ErrorActionPreference = "STOP"
 $ProgressPreference = "SilentlyContinue"
 
 # load module
 Import-Module (Join-Path $PSScriptRoot "\modules\SitecoreImageBuilder") -Force
 
 $tags = [System.Collections.ArrayList]@()
+
 $windowsVersionMapping = @{
     "1909"     = "1909"
     "1903"     = "1903"
     "ltsc2019" = "1809"
-    "1803"     = "1803"
 }
-foreach ($wv in $WindowsVersion)
+
+filter WindowsFilter { param([string]$Version) if($_ -like "*-windowsservercore-$($Version)" -or $_ -like "*-nanoserver-$($windowsVersionMapping[$Version])") { $_ } }
+filter SitecoreFilter { param([string]$Version) if($_ -like "*:$($Version)-windowsservercore-*" -or $_ -like "*:$($Version)-nanoserver-*") { $_ } }
+
+$availableSpecs = Get-BuildSpecifications -Path (Join-Path $PSScriptRoot "\windows")
+
+if(!$IncludeExperimental) {
+    Write-Host "Excluding experimental images."
+    $availableSpecs = $availableSpecs | Where-Object { !$_.Experimental }
+}
+
+$availableTags = $availableSpecs | Select-Object -ExpandProperty Tag
+$defaultTags = $availableTags | Where-Object { $_ -like "mssql-developer:*" -or $_ -like "sitecore-openjdk:*" }
+$miscTags = $availableTags | Where-Object { $_ -like "sitecore-certificates:*" -or $_ -like "sitecore-redis:*" }
```

Comment:
Is Redis required for XM/XP? I thought Redis was for XC only. Does this mean these images are built?